From ec118190123071e05f3ca5ed36651e7187341b4c Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Thu, 9 Apr 2015 16:57:08 -0700
Subject: [PATCH 7/7] upnpsoap: Don't honor the NO_RESIZE flag for thumbnails

Some images don't include an embedded thumbnail, and others
have the thumbnail in the wrong aspect ratio.  In these
cases we resize ourselves, which we should allow to occur
regardless of the NO_RESIZE flag.
---
 upnpsoap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/upnpsoap.c b/upnpsoap.c
index d9816a8..cc76c4c 100644
--- a/upnpsoap.c
+++ b/upnpsoap.c
@@ -620,7 +620,7 @@ add_resized_res(int srcw, int srch, int reqw, int reqh, char *dlna_pn,
 	int dstw = reqw;
 	int dsth = reqh;
 
-	if( args->flags & FLAG_NO_RESIZE )
+	if( (args->flags & FLAG_NO_RESIZE) && reqw > 160 && reqh > 160 )
 		return;
 
 	strcatf(args->str, "&lt;res ");
-- 
1.9.1

